polytracker/src/taintdag/polytracker.cpp

```diff
@@ -70,29 +70,40 @@ taint_range_t PolyTracker::create_source_taint(source_index_t src,
 //
 // If fd is not tracked as a taint source, no labels will be assigned to the
 // corresponding mem.
+// NOTE: If offset < 0, the assumption is made that offsets aren't known and
+// per source tracking should be used. This typically happens for stdin, sockets
+// and other streams.
 std::optional<taint_range_t> PolyTracker::source_taint(int fd, void const *mem,
                                                        source_offset_t offset,
                                                        size_t length) {
   if (auto source_index = output_file_.section<Sources>().mapping_idx(fd)) {
-    return this->create_source_taint(
-        *source_index, {reinterpret_cast<uint8_t const *>(mem), length},
-        offset);
+    if (offset < 0) {
```

Comment:
Ah, okay, I see what happens here. Hmm, yeah, ideally this would be handled in the various read hooks based on what functions like `ftell(int fd)` return. However that's not practical right now. Maybe this could be addressed the dfsan hook refactor?

Reply:
Sounds like a plan. Maybe we could just combine it in the same issue? I'll add it there and if you feel like it deserves a separate issue, just create one.

Reply:
Already referenced in #6515 .